baseplate/frameworks/pyramid/__init__.py

```diff
@@ -189,13 +190,14 @@ def _on_new_request(self, event: pyramid.events.ContextFound) -> None:
         if self.header_trust_handler.should_trust_edge_context_payload(request):
             try:
                 edge_payload = request.headers.get("X-Edge-Request", None)
+                edge_payload_decoded = base64.urlsafe_b64decode(edge_payload)
```

Comment:
This will throw a `TypeError` if `edge_payload` is `None`.  We might want to change this to:

```python
edge_payload = request.headers.get("X-Edge-Request", None)
if self.edge_context_factory and edge_payload is not None:
    edge_payload_decoded = base64.urlsafe_b64decode(edge_payload)
    edge_context = self.edge_context_factory.from_upstream(edge_payload_decoded)
    edge_context.attach_context(request)
```

Reply:
Also, as neil noted, this should be `b64decode`, not `urlsafe_`